### DISCUSSION TITLE
Testing component behaviour within the context of a given route

### DISCUSSION THREAD

**Discussion by paianoa** [livewire/livewire#3269] [1. Help]
Is it possible to test a component behaviour within a given route and its middlewares? AFAIK the only way to test is in "isolation mode" with the Livewire::test command ignoring the "route context" where the component is called/used. 

In my case I have a component, (re)used in different routes (and views) with their own middlewares, which behaves differently according to the middleware of the parent route but can't find a way to test this behaviour.

One solution would be to move middlewares inside the component but I prefer to keep middlewares references inside the routes/web.php file in order to have an instantly clear view of the middlewares used

**Comment by joshhanley** [comment]
@paianoa there is no way to run Livewire component tests at the route level, but we may be able to work it in a different way.

Can you be more specific as to what the different middleware do and how the Livewire component changes depending on these?
